cirq-core/cirq/sim/classical_simulator_test.py

```diff
@@ -0,0 +1,196 @@
+# Copyright 2023 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import numpy as np
+import pytest
+import cirq
+import sympy
+
+
+class TestSimulator:
+    def test_x_gate(self):
+        q0, q1 = cirq.LineQubit.range(2)
+        circuit = cirq.Circuit()
+        circuit.append(cirq.X(q0))
+        circuit.append(cirq.X(q1))
+        circuit.append(cirq.X(q1))
+        circuit.append(cirq.measure((q0, q1), key='key'))
+        expected_results = {'key': np.array([[[1, 0]]], dtype=np.uint8)}
+        sim = cirq.ClassicalStateSimulator()
+        results = sim._run(circuit=circuit, param_resolver=None, repetitions=1)
+        np.testing.assert_equal(results, expected_results)
+
+    def test_CNOT(self):
+        q0, q1 = cirq.LineQubit.range(2)
+        circuit = cirq.Circuit()
+        circuit.append(cirq.X(q0))
+        circuit.append(cirq.CNOT(q0, q1))
+        circuit.append(cirq.measure(q1, key='key'))
+        expected_results = {'key': np.array([[[1]]], dtype=np.uint8)}
+        sim = cirq.ClassicalStateSimulator()
+        results = sim._run(circuit=circuit, param_resolver=None, repetitions=1)
```

Comment:
`sim._run` is a private method. Please write the tests using the public method `sim.run`

Reply:
Since the function is a response to the _run function in the SimulatesStates class it wont's work correctly using the public run method